src/java/net/jpountz/lz4/LZ4BlockInputStream.java

```diff
@@ -147,7 +152,19 @@ public long skip(long n) throws IOException {
   }
 
   private void refill() throws IOException {
-    readFully(compressedBuffer, HEADER_LENGTH);
+    if (finished || o < originalLen) {
```

Comment:
Why is this check necessary? I don't find any execution path on which this check is evaluated to true.